config/v1/types_build.go

```diff
@@ -108,7 +108,7 @@ type BuildOverrides struct {
 
 	// NodeSelector is a selector which must be true for the build pod to fit on a node
 	// +optional
-	NodeSelector metav1.LabelSelector `json:"nodeSelector,omitempty"`
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
@deads2k i think we guessed wrong here.  From what I see pods only support map[string]string for a node selector value?  So why are we using a LabelSelector here?

If we do want a LabelSelector, how does the MatchExpression get mapped into the pod definition?

Reply:
/cc @adambkaplan

Reply:
I don't see a clear path to using `LabelSelector` cleanly unless upstream `Pod` plans on supporting `LabelSelector` for node selector.

Reply:
Hmm... honestly `metav1.LabelSelector` is supposed to be the future for all selectors, and getting from `LabelSelector` to `map[string]string` is as simple as calling `metav1.LabelSelectorAsMap`, so I'm tempted to leave this is as is since this opens gates for future. Although true, I doubt it will ever change for pod itself.

Reply:
my concern is it's going to lead to user confusion when they try to use matchexpressions and we ignore them.

if we want to introduce support for match expressions (or label selectors in general) in the future, we can always add a new api field and have it take priority over this one when set.

Reply:
@soltysh what does LabelSelectorAsMap do for matchexpression values?

Reply:
found my own answer to that:

```
Operators that cannot be converted into plain
labels (Exists, DoesNotExist, NotIn, and In with more than one value) will result in
an error.
```

so again i'd rather not lead users down a bad path.